db/rdbms/src/main/java/io/camunda/db/rdbms/write/service/FlowNodeInstanceWriter.java

```diff
@@ -36,13 +38,47 @@ public void create(final FlowNodeInstanceDbModel flowNode) {
             flowNode));
   }
 
-  public void end(final long flowNodeKey, final FlowNodeState state, final OffsetDateTime endDate) {
-    final var dto = new EndFlowNodeDto(flowNodeKey, state, endDate);
-    executionQueue.executeInQueue(
-        new QueueItem(
-            ContextType.FLOW_NODE,
-            flowNodeKey,
-            "io.camunda.db.rdbms.sql.FlowNodeInstanceMapper.updateStateAndEndDate",
-            dto));
+  public void end(
+      final long flowNodeInstanceKey, final FlowNodeState state, final OffsetDateTime endDate) {
+    final var dto = new EndFlowNodeDto(flowNodeInstanceKey, state, endDate);
+    final boolean wasMerged =
+        executionQueue.tryMergeWithExistingQueueItem(new EndFlowNodeToInsertMerger(dto));
+
+    if (!wasMerged) {
+      executionQueue.executeInQueue(
+          new QueueItem(
+              ContextType.FLOW_NODE,
+              flowNodeInstanceKey,
+              "io.camunda.db.rdbms.sql.FlowNodeInstanceMapper.updateStateAndEndDate",
+              dto));
+    }
+  }
+
+  public static class EndFlowNodeToInsertMerger implements QueueItemMerger {
+
+    private final EndFlowNodeDto dto;
+
+    public EndFlowNodeToInsertMerger(final EndFlowNodeDto dto) {
+      this.dto = dto;
+    }
+
+    @Override
+    public boolean canBeMerged(final QueueItem queueItem) {
+      return queueItem.contextType() == ContextType.FLOW_NODE
+          && queueItem.id().equals(dto.flowNodeInstanceKey())
+          && queueItem.parameter() instanceof FlowNodeInstanceDbModel;
+    }
+
+    @Override
+    public QueueItem merge(final QueueItem originalItem) {
+      final var newParameter =
+          FlowNodeInstanceDbModelBuilder.of((FlowNodeInstanceDbModel) originalItem.parameter())
```

Comment:
Hm, for UserTask i created a "toBuilder" function on the Model itself which creates the builder .. :D (At the end, same behaviour, but we should be consistent of how we create the builder. => Needs to be discussed.